waterlog/src/Components/HistoryTable.js

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from 'react';
+import { LowStatusIcon, MediumStatusIcon, HighStatusIcon } from './AlertBox';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { fetchSegmentsLeaksHistory } from '../actions/SegmentLeaksHistoryActions';
+import Loader from './Loader';
+
+export const getStatusIcon = function (severity) {
+	switch (severity.toLowerCase()) {
+		case 'high':
+			return HighStatusIcon();
+		case 'low':
+			return LowStatusIcon();
+		case 'medium':
+			return MediumStatusIcon();
+		default:
+			return null;
+	}
+};
+class HistoryTableComponent extends Component {
```

Comment:
Your component makes use of props.
Have a constructor that receives props as a parameter and make a call to the super function having a props parameter too.
```
constructor(props)
{
    super(props)
}
```

Reply:
Done